c++/src/kj/compat/tls.h

```diff
@@ -97,6 +101,9 @@ class TlsContext {
 
     kj::Maybe<kj::Duration> acceptTimeout;
     // Timeout applied to accepting a new TLS connection. `timer` is required if this is set.
+
+    kj::Maybe<TlsErrorHandler&> acceptErrorHandler;
```

Comment:
Hmm, it's unusual to take a `kj::Function` by reference like this, and I worry it's going to lead to some awkward code and maybe UAF potential as people don't realize they have to keep the function object itself alive.

Did you find that a lot of stuff breaks if `Options` is converted to move-only semantics?

Reply:
> Hmm, it's unusual to take a kj::Function by reference like this, and I worry it's going to lead to some awkward code and maybe UAF potential as people don't realize they have to keep the function object itself alive.

That's a fair concern. I'm guessing you'd consider `kj::Function` a value type? And pass by reference is only appropriate for resource types?

> Did you find that a lot of stuff breaks if Options is converted to move-only semantics?

I think our production code itself builds with move-only semantics but it broke a test or two. If you're okay with trivially breaking some downstream builds here, I'm fine making the change. I can also make a nested class refcounted wrapper if you'd prefer that.